applications/asset_tracker_v2/doc/location_module.rst

```diff
@@ -0,0 +1,117 @@
+
+* GNSS set as the first priority method and cellular positioning as the second priority.
+  Used location methods can be configured at build time
+  (:ref:`CONFIG_DATA_SAMPLE_GNSS_DEFAULT <CONFIG_DATA_SAMPLE_GNSS_DEFAULT>` and
+  :ref:`CONFIG_DATA_SAMPLE_NEIGHBOR_CELLS_DEFAULT <CONFIG_DATA_SAMPLE_NEIGHBOR_CELLS_DEFAULT>`)
+  or to the dynamically updated value set by the cloud side and stored to flash.
+* Timeout set to either the value configured at build time
+  (:ref:`CONFIG_DATA_LOCATION_TIMEOUT_SECONDS <CONFIG_DATA_LOCATION_TIMEOUT_SECONDS>`)
+  or to the dynamically updated value set by the cloud side and stored to flash.
+  Values stored in flash take precedence.
+
+The location module receives configuration updates as payload in the following two events:
+
+* ``DATA_EVT_CONFIG_INIT`` - The event contains the value configured at build time or received from cloud and stored to flash.
+  The event is distributed from the data module as part of the application initialization.
+* ``DATA_EVT_CONFIG_READY`` - The event is received when the data module has processed an incoming message from cloud with a configuration update.
+
+When a position has been retrieved, the ``LOCATION_MODULE_EVT_DATA_READY`` event is sent from the module.
+If location request failed, ``LOCATION_MODULE_EVT_DATA_NOT_READY`` is sent.
+If a timeout occurred, ``LOCATION_MODULE_EVT_TIMEOUT`` is sent.
+
+LNA configuration
+=================
+
+Different devices have different antenna and LNA setups depending on the antenna type (onboard or external).
+The application uses the :ref:`lib_modem_antenna` library for configuring the LNA.
+The library has Kconfig options to control the LNA using either the COEX0 or MAGPIO interface of the nRF9160.
+See the library documentation for more details on how to configure the antenna and LNA.
+
+GPS assistance data
+===================
+
+The location module receives requests for GPS assistance data from the :ref:`lib_location` library.
+When an A-GPS request is received, the module distributes it to the other modules as a ``LOCATION_MODULE_EVT_AGPS_NEEDED`` event that contains information about the type of assistance data needed.
+Providing the requested A-GPS data may reduce the time it takes to acquire a GNSS fix.
+
+Module internals
+================
+
+The location module is a threadless module in terms of message processing.
+In this sense, it differs from the other modules.
+It has a thread for processing location events from the :ref:`lib_location` library, whereas other modules use a thread for processing messages.
+The thread is used to offload data processing from the location callbacks coming from the :ref:`lib_location` library, and no time consuming tasks should be performed there.
+
+All incoming events from other modules are handled in the context of the Application Event Manager callback, as they all complete fast enough to not require a separate thread.
```

Comment:
```suggestion
All incoming events from other modules are handled in the context of the Application Event Manager callback, because they all complete fast enough to not require a separate thread.
```